### PR TITLE
Use factory_boy for text fixtures.

### LINKED ISSUE #331: Use pytest-fatoryboy in tests for data fixtures
While working on #330 I found the way we handle data fixture not enough and think we would profit from using data factories for the test data. Luckily there is [pytest-fatoryboy](https://github.com/pytest-dev/pytest-factoryboy), [a good article describes how it works](http://developer.paylogic.com/articles/factory-injection-combining-pytest-factory_boy.html).

### PR DESCRIPTION
**Pull request by jezdez**
This uses factory_boy and pytest-factoryboy for easier text fixtures.

Fix #331.

### CODE REVIEW

**Comment by jezdez** [comment]
This isn't complete yet, the dashboard tests are still missing.

File: tests/test_dashboard.py [hunk 1]
```diff
@@ -58,44 +50,24 @@ def make_cluster(mocker, **kwargs):
...
-        **kwargs
+    ClusterFactory.create_batch(
+        5,
+        created_by=user,
+        most_recent_status=Cluster.STATUS_WAITING,
     )
-
-
-@pytest.fixture
-def dashboard_clusters(mocker, now, test_user, ssh_key):
```
Comment (jezdez) [inline tests/test_dashboard.py]: @robhudson Check out how factory_boy works with batched object creations..

Comment (robhudson) [inline tests/test_dashboard.py]: 👍

File: atmo/clusters/factories.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from ..keys.factories import SSHKeyFactory
+from ..users.factories import UserFactory
+
+
+class ClusterFactory(factory.django.DjangoModelFactory):
+    identifier = factory.Sequence(lambda n: 'test-cluster-%s' % n)
+    size = 5
+    ssh_key = factory.SubFactory(SSHKeyFactory)
+    # start_date
+    # end_date = None
```
Comment (jezdez) [inline atmo/clusters/factories.py]: Lemme uncomment it, I wanted to have the attributes defined.

**Pull request merged**